hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -31,48 +32,77 @@ class SapHanaCollector(object):
     SAP HANA database data exporter
     """
 
-    def __init__(self, conector):
+    def __init__(self, connector):
         super(SapHanaCollector, self).__init__()
         self._logger = logging.getLogger(__name__)
-        self._hdb_connector = conector
+        self._hdb_connector = connector
 
-    def _execute(self, metric):
+    def _execute(self, query):
         """
-        Create metric object
+        Execute query and return a QueryResult object
 
         Args:
-            metric (dict): query, info, type structure dictionary
+            query (string): SQL query
         """
         try:
-            value = self._hdb_connector.query(metric['query'])
-
-            if metric['type'] == core.GaugeMetricFamily:
-                metric_obj = self._manage_gauge(metric, value)
-            else:
-                raise NotImplementedError('{} type not implemented'.format(metric['type']))
-
-            return metric_obj
+            query_result = self._hdb_connector.query(query)
+            return query_result
         except KeyError as err:
             raise MalformedMetric(err)
 
-    def _manage_gauge(self, metric, value):
+    def _format_query_result(self, query_result):
+        """
+        Format query results to match column names with their values for each row
+        Returns nested list, containing tuples (column_name, value)
+
+        Args:
+            query_result (obj): QueryResult object
+        """
+        query_columns = []
```

Comment:
I don't like the result format. We should avoid to use tuples here. They are not really convenient. They forces us to use indexes to manage the data and is not so readable.
We should use `named tuples`, dictionaries or just create simple class to manage that.
We will appreciate it in the future